server/service/src/report/report_service.rs

```diff
@@ -325,16 +324,88 @@ fn get_report(ctx: &ServiceContext, id: &str) -> Result<Report, RepositoryError>
 
 fn query_reports(
     ctx: &ServiceContext,
-    pagination: Option<PaginationOption>,
     filter: Option<ReportFilter>,
     sort: Option<ReportSort>,
 ) -> Result<Vec<Report>, ListError> {
+    let app_version: Version = Version::from_package_json();
+
     let repo = ReportRepository::new(&ctx.connection);
-    let pagination = get_default_pagination(pagination, MAX_LIMIT, MIN_LIMIT)?;
     let filter = filter
         .unwrap_or_default()
         .r#type(ReportType::OmSupply.equal_to());
-    Ok(repo.query(pagination, Some(filter.clone()), sort)?)
+
+    let reports_to_show_meta_data = report_filter_method(
```

Comment:
I think it's better to return report_ids to show (that feels a bit more clear to me), maybe it's personal preference, how does it sound to you ?

Reply:
Yes I like that too - will amend

Reply:
1ce0ac42e7de686b25dbad610cf2a192f3a90f3a